File: youlexuan-service-interface/src/main/java/com/youlexuan/sellergoods/service/BrandService.java
package com.youlexuan.sellergoods.service;

import com.youlexuan.entity.PageResult;
import com.youlexuan.pojo.TbBrand;

import java.util.List;
import java.util.Map;

public interface BrandService {
    //显示所有品牌商品
    public List<TbBrand> findAll();

    //分页+模糊查询
    PageResult findPage(TbBrand brand, int pageNum,int pageSize);

    //添加品牌
    public void add(TbBrand brand);

    //修改品牌
    public void update(TbBrand brand);

    //修改品牌前需要先查出来
    TbBrand findOne(Long id);

    //批量删除
    public void delet(Long[] ids);

    //下拉列表从数据库中获取
    List<Map> selectOption();
}

File: youlexuan-page-web/src/main/webapp/js/service/brandService.js
app.service('brandService',function ($http) {

    //分页
    this.findPage=function (pageNum,pageSize) {
        return $http.get('../brand/findPage.do?page='+pageNum+'&rows='+pageSize);
    }

    /**
     * @param pageNum
     * @param pageSize
     * 根据条件模糊查询
     */
    this.search = function (pageNum,pageSize,searchEntiy) {
        return $http.post('../brand/search.do?page='+pageNum+'&rows='+pageSize,searchEntiy);
    }

    this.save=function (entity) {
        var mname="add.do";
        if (entity.id != null){
            mname="update.do";
        }
        return $http.post('../brand/'+mname,entity);
    }

    this.findOne=function (id) {
        return $http.get('../brand/findOne.do?id='+id);
    }

    //批量删除
    this.delet=function () {
        //获取选中的复选框
        return $http.get('../brand/delet.do?ids='+selectIds);
    }

    this.selectOption=function () {
        return $http.get('../brand/selectOption.do')
    }

})